axon-framework/serialization.md

```diff
@@ -27,7 +37,22 @@ To that end, the `Configuration` API allows you to define default, message and e
    These objects are generally not shared between different applications, and most of these classes aren't expected to have some of the getters and setters that are, for example, typically required by Jackson based serializers.
    A flexible, general purpose serializer like [XStream](http://x-stream.github.io/) is quite suited for this purpose.
 
-By default, all three `Serializer` flavors are set to use the `XStreamSerializer`, which internally uses [XStream](http://x-stream.github.io/) to serialize objects to an XML format. XML is a verbose format to serialize to, but XStream has the major benefit of being able to serialize virtually anything. This verbosity is typically fine when storing tokens, sagas or snapshots, but for messages \(and specifically events\) XML might prove to cost too much due to its serialized size. Thus for optimization reasons you can configure different serializers for your messages. Another very valid reason for customizing serializers is to achieve interoperability between different \(Axon\) applications, where the receiving end potentially enforces a specific serialized format.
+By default, all three `Serializer` flavors are set to use the `XStreamSerializer`, which internally uses [XStream](http://x-stream.github.io/) to serialize objects to an XML format. 
+XML is a verbose format to serialize to, but XStream has the major benefit of being able to serialize virtually anything.
+
+> *XStream and JDK 17*
+> 
+> Although XStream can "serialize virtually anything," more recent versions of the JDK impede its flexibility.
+> This predicament comes down to XStream's reflective approach to finding out how to de-/serialize *any* object, which has become problematic with Java's intent to secure its internals.
+> Hence, if you're using JDK 17, the chances are that objects (e.g., your sagas) intended for serialization require additional configuration.
+> 
+> On some occasions setting XStream's security settings is sufficient.
+> Other times you will have to introduce custom [`Converters`](https://x-stream.github.io/converters.html) to de-/serialize specific types.
+> If you prefer not to deal with specific XStream settings, it might be a good step to use the `JacksonSerializer` throughout your Axon application.
```

Comment:
```suggestion
> If you prefer not to deal with specific XStream settings, it might be better to use the `JacksonSerializer` throughout your Axon application.
```